docs/rfd/002.md

```diff
@@ -0,0 +1,94 @@
+32 raw octets. This causes a variety of problems, e.g. when transferring keys
+between hosts or when specifying server keys for *GLOME Login*.
+
+See also [google/glome#100](https://github.com/google/glome/issues/100).
+
+# Requirements
+
+* A GLOME public key at rest should be unambiguously identifiable as such.
+* Public keys should be printable.
+* Public keys should be easily exchanged over any medium, potentially analog.
+* The public key format should allow associating additional context with a key.
+
+# Design ideas
+
+Public keys are stored in base64 encoding and tagged with their protocol
+variant. The configuration file format accepts keys in a format similar to
+[OpenSSH's `authorized_keys` format][1].
+
+[1]: https://man.openbsd.org/sshd.8#AUTHORIZED_KEYS_FILE_FORMAT
+
+## Public Key Format
+
+The format of a _GLOME public key_ adheres to the ABNF below:
+
+```abnf
+public-key = key-type SP key-base64
+key-type = "glome-x25519-sha256"
+key-base64 = 44base64-char
+base64-char = "=" / "+" / "/" / ALPHA / DIGIT
+```
+
+The key type encodes the GLOME variant this key should be used with. As we
+only have one variant right now, we're only defining one `key-type` here.
+Future variants should follow the naming pattern `glome-<DHALG>-<HMACALG>`.
+
+An example public key, like it should be created by `glome pubkey`:
+
+```
+glome-x25519-sha256 lXmlq5jynG6um/w4D4N13TRIE+x7jt0TKVNDMSRS2/I=
+```
+
+## Public Key Interpretation
+
+An implementation must verify that the `key-type` matches its expectations and
+must not produce a tag if it does not.
+
+If the `key-type` matches the expectations, the `key-base64` part is decoded as
+base64, and the resulting 32 octets are interpreted as the _raw GLOME public
+key_, suitable for use with `glome_tag`.
+
+## Consequences for the GLOME Login Configuration Format
+
+The GLOME login configuration file consists of a number of public keys,
+one per line, and optional comments. Specifying a client key in the config file
+is not supported anymore. All other existing configuration items are specified
+as per-key options (i.e. `key-version` and `url-prefix`).
+
+The login client should render a challenge for each of the configured server
+public keys. It should allow access if the entered response matches any of the
+presented challenges. For environments with a single server key, this behaviour
+is the same as for the old config file format.
+
+The ABNF specification of the config file is given below:
+
+```abnf
+config-file = *(config-line LF)
+config-line = annotated-public-key / comment
+annotated-public-key = [ *option SP ] public-key [ SP inline-comment ]
+option = option-key "=" option-value 
+option-key = 1*(ALPHA / DIGIT / "-")
+option-value = *%x21-7e
+comment = "#" inline-comment
+inline-comment = *%x20-7e
+```
+
+An example config file:
+
+```
+key-version=4 url-prefix=https://glome.example.com glome-x25519-sha256 lXmlq5jynG6um/w4D4N13TRIE+x7jt0TKVNDMSRS2/I= standard authorization server
```

Comment:
It seems like this last version of the proposal is maybe closer to the original ini file rather than the authorized_keys. Perhaps implementing a simple ini parser from scratch maybe wasn't such a bad idea in the end? :)

Reply:
I'll remove the changes for the config file format after #112 has landed (thanks, btw). We can have the discussion of multi-key support separately from the discussion about public key format.